src/utils/getAnsibleMetaData.ts

```diff
@@ -0,0 +1,197 @@
+      );
+      return undefined;
+    }
+  } catch (error) {
+    console.log(
+      `cmd '${cmd} ${arg}' was not executed with the following error: ' ${error.toString()}`
+    );
+    return undefined;
+  }
+
+  return result;
+}
+
+async function getAnsibleInfo() {
+  const ansibleInfo = {};
+
+  const ansibleVersionObj = (await context.ansibleConfig).ansible_meta_data;
+  const ansibleVersionObjKeys = Object.keys(ansibleVersionObj);
+
+  // return empty if ansible --version fails to execute
+  if (ansibleVersionObjKeys.length === 0) {
+    return ansibleInfo;
+  }
+
+  let ansibleVersion;
+  if (ansibleVersionObjKeys[0].includes(" [")) {
+    ansibleVersion = ansibleVersionObjKeys[0].split(" [");
+  } else {
+    ansibleVersion = ansibleVersionObjKeys[0].split(" ");
+  }
+  ansibleInfo["ansible version"] = `Ansible ${ansibleVersion[1].slice(0, -1)}`;
+
+  ansibleInfo["ansible location"] = (
+    await context.ansibleConfig
+  ).ansible_location;
+
+  ansibleInfo["config file path"] = [ansibleVersionObj["config file"]];
+
+  ansibleInfo["ansible collections location"] = (
+    await context.ansibleConfig
+  ).collections_paths;
+
+  ansibleInfo["ansible module location"] = (
+    await context.ansibleConfig
+  ).module_locations;
+
+  ansibleInfo["ansible default host list path"] = (
+    await context.ansibleConfig
+  ).default_host_list;
+
+  // console.log("*** ansible info -> ", ansibleInfo);
+  return ansibleInfo;
+}
+
+async function getPythonInfo() {
+  const pythonInfo = {};
+
+  const pythonVersionResult = await getResultsThroughCommandRunner(
+    "python",
+    "--version"
+  );
+  if (pythonVersionResult === undefined) {
+    return pythonInfo;
+  }
+
+  pythonInfo["python version"] = pythonVersionResult.stdout.trim();
+
+  const pythonPathResult = await getResultsThroughCommandRunner(
+    "python",
+    '-c "import sys; print(sys.executable)"'
+  );
+  pythonInfo["python location"] = pythonPathResult.stdout.trim();
+
+  // console.log("*** python info -> ", pythonInfo);
+  return pythonInfo;
+}
+
+async function getAnsibleLintInfo() {
+  const ansibleLintInfo = {};
+
+  const ansibleLintVersionResult = await getResultsThroughCommandRunner(
+    "ansible-lint",
+    "--version"
+  );
+
+  if (ansibleLintVersionResult === undefined) {
+    return ansibleLintInfo;
+  }
+
+  const ansibleLintPathResult = await getResultsThroughCommandRunner(
+    "which",
+    "ansible-lint"
+  );
+
+  ansibleLintInfo["ansible-lint version"] =
+    ansibleLintVersionResult.stdout.trim();
+
+  ansibleLintInfo["ansible-lint location"] =
+    ansibleLintPathResult.stdout.trim();
+
+  // console.log("*** ansible-lint info -> ", ansibleLintInfo);
+  return ansibleLintInfo;
+}
+
+async function getExecutionEnvironmentInfo() {
+  const eeInfo = {};
+
+  const basicDetails = (await context.executionEnvironment)
+    .getBasicContainerAndImageDetails;
+
+  eeInfo["container engine"] = basicDetails.containerEngine;
```

Comment:
Do we really need to use spaces here? I wonder if it would not be easier for us to use short names and avoid repeating things, like:

`image`, `image_id`, `volume_mounts`?

Reply:
I kept it this way because it becomes easier for the client to directly use the key to set it as the label in the UI.